builder/osc/common/step_run_source_vm.go

```diff
@@ -108,25 +108,31 @@ func (s *StepRunSourceVm) Run(ctx context.Context, state multistep.StateBag) mul
 		return multistep.ActionHalt
 	}
 
-	subregion := state.Get("subregion_name").(string)
+	blockDevice := s.BlockDevices.BuildOSCLaunchDevices()
 	vmcount := int32(1)
+	subregion := state.Get("subregion_name").(string)
+	subnetID := state.Get("subnet_id").(string)
+
 	runOpts := oscgo.CreateVmsRequest{
-		ImageId:      s.SourceOMI,
-		VmType:       &s.VmType,
-		UserData:     &userData,
-		MaxVmsCount:  &vmcount,
-		MinVmsCount:  &vmcount,
-		Placement:    &oscgo.Placement{SubregionName: &subregion},
-		BsuOptimized: &s.BsuOptimized,
-		//BlockDeviceMappings: s.BlockDevices.BuildOSCLaunchDevices(),
+		ImageId:             s.SourceOMI,
+		VmType:              &s.VmType,
+		UserData:            &userData,
+		MaxVmsCount:         &vmcount,
+		MinVmsCount:         &vmcount,
+		BsuOptimized:        &s.BsuOptimized,
+		BlockDeviceMappings: &blockDevice,
+	}
+
+	log.Printf("subregion is %s", subregion)
+	if subregion != "" {
+		log.Printf("placement subregion is %s", subregion)
```

Comment:
Duplicate log, you can remove one of them

Reply:
removed